tensorstore/kvstore/s3/credentials/BUILD

```diff
@@ -0,0 +1,115 @@
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+load("//bazel:tensorstore.bzl", "tensorstore_cc_library", "tensorstore_cc_test")
+
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])
+
+tensorstore_cc_library(
```

Comment:
I think that you could break this into separate libraries if you did the following:

1. aws_credential_provider.h has only: AwsCredentials, AwsCredentialProvider.

2. default_credential_provider  adds: 

AwsCredentialProviderFn
RegisterAwsCredentialProviderProvider
GetAwsCredentialProvider

3. Users would mostly need to include awsx_credential_provider; the driver would need to include default_credential_provider as well.

Reply:
This has been separated into the following libraries: 

1. aws_credential_provider
2. default_credentials_provider
3. environment_credentials_provider
4. file_credentials_provider
5. ec2_credentials_provider